### PR TITLE
feat: allow aggregations without group bys

### LINKED ISSUE #430: Cannot create aggregate without a GROUP BY
KSQL insists on having a `GROUP BY` column, even if logically it makes sense without. 

For example: how many tweets were there in each hourly period? 

```
ksql> select count(*) from twitter window tumbling (size 1 hour);
Aggregate query needs GROUP BY clause.
ksql>
```

### PR DESCRIPTION
**Pull request by nateab**
### Description 
Helps to partially fix #430 by allowing users to create aggregations without group by clauses in push queries .

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java [hunk 1]
```diff
@@ -668,8 +671,11 @@ private void throwOnUdafs(final Expression expression) {
         public Void visitFunctionCall(final FunctionCall functionCall, final Void context) {
           final FunctionName functionName = functionCall.getName();
           if (metaStore.isAggregate(functionName)) {
-            throw new KsqlException("Use of aggregate function "
-                + functionName.text() + " requires a GROUP BY clause.");
+            analysis.addAggregateFunction(functionCall);
```
Comment (agavra) [inline ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java:674]: we should probably change the method name, right?

File: ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java [hunk 2]
```diff
@@ -668,8 +671,11 @@ private void throwOnUdafs(final Expression expression) {
...
+            // Since this is a dummy group by, we don't actually need a correct node location
+            final NodeLocation nodeLocation = new NodeLocation(0,0);
```
Comment (agavra) [inline ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java:676]: can we just use `Optional.empty()` here?

File: ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/push-queries.json [hunk 3]
```diff
@@ -653,6 +653,26 @@
         ]}
       ]
     },
+    {
+      "name": "aggregate without GROUP BY expression",
```
Comment (agavra) [inline ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/push-queries.json:657]: can we add tests for (1) what happens if you try to `CREATE TABLE` without a group by (basically confirm a good error message), (2) what happens if you select a non-grouping column (e.g. `SELECT COUNT(*), ID`) and (3) that multiple aggregate functions work?

File: ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java [hunk 4]
```diff
@@ -668,8 +671,11 @@ private void throwOnUdafs(final Expression expression) {
...
+            analysis.setGroupBy(new GroupBy(Optional.of(nodeLocation),
+                ImmutableList.of(new IntegerLiteral(1))));
```
Comment (agavra) [inline ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java:678]: thoughts on using just a single byte to save some space?

**Review by agavra — COMMENTED** [review]
Thanks Natea - this looks way simpler than I thought it would be! that's pretty elegant. I think we should try to make sure the error message is super clear if you try to `CREATE TABLE` without a GROUP BY but with an aggregate function (e.g. it should say "not supported without GROUP BY" instead of "must select key").

File: ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java (hunk 1 above)
Comment (nateab) [reply ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java:674]: yes good catch!

File: ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java (hunk 2 above)
Comment (nateab) [reply ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java:676]: added

File: ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java (hunk 4 above)
Comment (nateab) [reply ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java:678]: switched to BytesLiteral

File: ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/push-queries.json (hunk 3 above)
Comment (nateab) [reply ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/push-queries.json:657]: I will add tests covering cases 1 and 3.

For 2, this is what currently happens 
```
ksql> select avg_stars,count(*) from RATINGS_TABLE EMIT CHANGES;
Non-aggregate SELECT expression(s) not part of GROUP BY: AVG_STARS
Either add the column(s) to the GROUP BY or remove them from the SELECT.
```
Is that reasonable?

Comment (nateab) [reply ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/push-queries.json:657]: discussed offline, confirmed this was reasonable

Comment (hli21) [reply ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/push-queries.json:657]: How about a stream pull query case, like "SELECT count(column) AS count FROM test"?

Comment (nateab) [reply ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/push-queries.json:657]: I have manually tested the stream pull query case, but I can also add an explicit test case for it too

Comment (nateab) [reply ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/push-queries.json:657]: Actually pull queries dont support group by clauses at all, unless you mean add a test to make sure we get the expected error? This is the current error
```
Pull queries don't support GROUP BY clauses. See https://cnfl.io/queries for more info.
Add EMIT CHANGES if you intended to issue a push query.
Statement: select count(*), sum(user_id) from RATINGS_STREAM;
```

**Review by agavra — APPROVED** [review]
Nice! Thanks @nateab

**Pull request merged**